language/move-prover/src/spec_translator.rs

```diff
@@ -901,6 +906,20 @@ impl<'env> SpecTranslator<'env> {
         }
     }
 
+    fn translate_pack(&self, args: &[Exp]) {
+        emit!(
+            self.writer,
+            "Vector({}EmptyValueArray",
+            "ExtendValueArray(".repeat(args.len())
```

Comment:
Nice! This trick avoids recursion for doing this.

Reply:
Yes! :)